src/Pate/Verification/Widening.hs

```diff
@@ -277,21 +301,40 @@ abstractOverVars scope_pre bundle _from _to postSpec postResult = withSym $ \sym
         True <- lift $ isPredTrue' heuristicTimeout (PS.unSE asm)
         return se'
 
-    PS.scopedExprMap sym postResult $ \(se :: PS.ScopedExpr sym tp pre) -> do
-      mres <- runMaybeT $
-            asScopedConst (W4.truePred sym) se
-        <|> asStackOffset PBi.OriginalRepr se
-        <|> asStackOffset PBi.PatchedRepr se
-        <|> asSimpleAssign se
-      case mres of
-        Just se' -> return se'
-        -- FIXME: add graceful failure mode
-        Nothing -> do
+      doRescope :: forall tp l. PL.Location sym arch l -> PS.ScopedExpr sym pre tp -> EquivM_ sym arch (MaybeF (PS.ScopedExpr sym post) tp)
+      doRescope _loc se = W4B.idxCacheEval cache (PS.unSE se) $ runMaybeTF $ do
+          se' <- (    asConcrete se
+                  <|> asStackOffset PBi.OriginalRepr se
+                  <|> asStackOffset PBi.PatchedRepr se
+                  <|> asScopedConst (W4.truePred sym) se
+                  <|> asSimpleAssign se
+                 )
```

Comment:
You hinted that the order of these might be important - can we document the rationale, if so?